### PR TITLE
Implement sorting functionality

### LINKED ISSUE #113: Sort search results
**Story:** As a user I want to sort search resalts by price or posted time in order to get increasing or dekreasing list of ads according search criteria.

**DoR:**
- [ ] The application works before the changes takes place 
- [ ] In the application are items (ad) stored in the database for testing purpose

**Acceptance criteria:**
- [ ] AC1: Dropdown with following sort criteria is available on the page with search results:
               EN: Highest price, Lowest price, Most recent, Oldest (most reduced). --> DGU: place is also a sort criteria for sort up/down direction
               DE: Teuerste, Günstigste, Älteste, Neueste, Ort alphab aufsteigend, Ort albabeb absteigend
- [ ] AC2: Default: most recent.  --> DGU changed at 2022.09.28: the default today is unknown. The default ist **place ascending** This is the default value for initial show results. 
Example with data of 2022.09.28 on INT environment (picture is placed in the comments): 
All ad in order 
Geneve
Genf
Kreuzlingen
Solothurn
Thusis
Worblaufen

- [ ] AC3: Sort by the highest price show the ad with highest price as a first one and futher decreasing to the ad with lowest price (for request "max rent" should be use,  no value = 0)
- [ ] AC4: Sort by the lowest price show the ad with lowest price as a first one and futher increasing to the ad with highest price
- [ ] AC5: Sort by the most recent show the last published (newest) ad as a first one and futher decreasing to the ad a latest publish date
- [ ]  AC6: Sort by the oldest show at latest point of time published ad as a first one and futher increasing to the last published ad (newest)
- [ ] AC7: When the user setting "language" is german, then the elements in the dropdown list are titled in german vs. when english then titled in english. 

**DoD:**
- [ ] The new dropdown item is selectable
- [ ] When the dropdown item gets changed, the resultset will be reloaded with the order selected in the dropdown list
- [ ] The title (dgu 29.09.2022 change: not title. place ascending) of the ad is the default search parameter and is used at every initial run to show the results 

Check how the fields for search and creation are added to the forms - subcategory_field, field and field_options tables in the database(maybe add a sortable column and sortable_options?).

### PR DESCRIPTION
**Pull request by asiday**
Chnages:
-Add initaial sorting by location
-Add histry styte for the order, that is unchanged after visiting specific accommodation

Close #113

### CODE REVIEW

**Review by T-z — APPROVED** [review]

File: am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx [hunk 1]
```diff
@@ -4,27 +4,58 @@ import SearchResultItem from './SearchResultItem.jsx';
...
   }, [props.language]);
 
+  /**
+   * This Hook is used to set property result after render.
+   * Also, it executes sort method when the component is initially rendered or updated,
+   * without losing the sort type, here it is named 'order'.
+   */
   useEffect(() => {
     setResults(props.results);
+    sortResult(history.state.order, props.results);
```
Comment (asiday) [inline am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx:23]: Is it a Hook?

Comment (asiday) [reply am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx:23]: Yes

File: am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx [hunk 2]
```diff
@@ -4,27 +4,58 @@ import SearchResultItem from './SearchResultItem.jsx';
...
     i18n.changeLanguage(props.language);
   }, [props.language]);
 
+  /**
+   * This Hook is used to set property result after render.
+   * Also, it executes sort method when the component is initially rendered or updated,
+   * without losing the sort type, here it is named 'order'.
+   */
   useEffect(() => {
     setResults(props.results);
```
Comment (asiday) [inline am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx:22]: What do you by hook?

Comment (asiday) [reply am-ui/webcomponents/search-results-webcomponent/src/SearchResultList.jsx:22]: dfsf

**Review by orrecx — APPROVED** [review]

**Pull request merged**